boot/bootutil/src/bootutil_misc.c

```diff
@@ -309,13 +309,13 @@ boot_write_trailer(const struct flash_area *fap, uint32_t off,
     int rc;
 
     align = flash_area_align(fap);
-    if (inlen > BOOT_MAX_ALIGN || align > BOOT_MAX_ALIGN) {
+    if (inlen > BOOT_MAX_ALIGN || align > BOOT_MAX_ALIGN ||
+	(BOOT_MAX_ALIGN & (align - 1)) == 0) {
```

Comment:
It is `!=`, not `==`, also there is tab here, MCUboot uses spaces, please fix, same applies on the other file.